### PR TITLE
Add BNPL flow [22110]

### PR DESCRIPTION
**Pull request by lqgioan**
## Description
- Added the code to support BNPL flow on mobile(encode, decode base64 message payload).
- Update post message payload for BNPL.

## Motivation & Context
Ticket: https://app.shortcut.com/credify/story/22110/feature-ios-servicex-add-bnpl-flow

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Credify/Credify/Objects/AppState.swift [hunk 1]
```diff
@@ -18,4 +18,5 @@ class AppState {
     var pushClaimTokensTask: ((String, ((Bool) -> Void)?) -> Void)? = nil
     var redemptionResult: ((RedemptionResult) -> Void)? = nil
     var dismissCompletion: (() -> Void)? = nil
+    var bnplEedemptionResult: ((_ status: RedemptionResult,_ orderId: String,_ isPaymentCompleted: Bool) -> Void)? = nil
```
Comment (ngo275) [inline Credify/Credify/Objects/AppState.swift]: typo

File: Credify/Credify/Objects/Entities.swift [hunk 2]
```diff
@@ -273,3 +335,235 @@ public enum ProductType : String {
 //    case consumerBNPL = "consumer"
 //    case merchantBNPL = "merchant"
 }
+
+/// Reference on this site https://app.quicktype.io/#l=swift
+public class AnyValue: Codable {
```
Comment (ngo275) [inline Credify/Credify/Objects/Entities.swift]: Please separate files. This file should focus on Credify related objects only.

File: Credify/Credify/serviceX.swift [hunk 3]
```diff
@@ -166,20 +165,33 @@ public struct serviceX {
...
+                completedBnplProviders: completedBnplProviders
+            )
             let vc = WebViewController.instantiate(context: context)
             let navigationController = UINavigationController(rootViewController: vc)
             navigationController.modalPresentationStyle = .overFullScreen
             navigationController.interactivePopGestureRecognizer?.isEnabled = false // disable navigation bar swipe back
             from.present(navigationController, animated: true)
         }
+        
+        public func getCompletedBnplProvider(
```
Comment (ngo275) [inline Credify/Credify/serviceX.swift]: `getConnectedBnplProviders`

File: Credify/Credify/serviceX.swift [hunk 4]
```diff
@@ -166,20 +165,33 @@ public struct serviceX {
         ///   - pushClaimTokensTask: A task that calls your push claim token API. This SDK needs to receive success status of this task.
         ///   - completionHandler: Completion handler. You can get notified about the result of the BNPL flow.
         public func presentModally(from: UIViewController,
-                                   offer: OfferData,
+                                   offers: [OfferData],
```
Comment (ngo275) [inline Credify/Credify/serviceX.swift]: https://github.com/credify-pte-ltd/credify-ios-sdk-kit/pull/4#pullrequestreview-934449915

File: Credify/Credify/Objects/AppState.swift (hunk 1 above)
Comment (lqgioan) [reply Credify/Credify/Objects/AppState.swift]: I corrected the function name

File: Credify/Credify/Objects/Entities.swift (hunk 2 above)
Comment (lqgioan) [reply Credify/Credify/Objects/Entities.swift]: Moved it to `CommonModel` file

File: Credify/Credify/serviceX.swift (hunk 3 above)
Comment (lqgioan) [reply Credify/Credify/serviceX.swift]: @ngo275 I updated it

File: Credify/Credify/serviceX.swift (hunk 4 above)
Comment (lqgioan) [reply Credify/Credify/serviceX.swift]: I'm checking it

Comment (lqgioan) [reply Credify/Credify/serviceX.swift]: @ngo275  I compiled `getOffers` and `getConnectedProviders` into on method that named with `getOffersAndConnectedProviders `

Comment (ngo275) [reply Credify/Credify/serviceX.swift]: @lqgioan 
`getOffersAndConnectedProviders` is too much information. All marketplaces should know is `getBNPLAvailability`, I would say. Also, the marketplace should know whether BNPL button should be hidden or not. So `getBNPLAvailability` should return a flag that says BNPL is available for users on the marketplaces.

Comment (lqgioan) [reply Credify/Credify/serviceX.swift]: @ngo275 I fixed and pushed the code

File: Credify/Credify/Objects/PostMessageModel.swift [hunk 5]
```diff
@@ -26,4 +26,27 @@ internal enum ReceiveMessageHandler: String {
...
     case offerTransactionStatusChanged
     case actionClose
+    case bnplPaymentComplete
+}
+
+internal class StartBnplMessage: Codable {
+    let offers: [OfferData]
+    let profile: CredifyUserModel
+    let orderId: String
+    let completeBnplProviders: [Organization]
```
Comment (ngo275) [inline Credify/Credify/Objects/PostMessageModel.swift:36]: `completeBnplProviders` is correct? Not `completedBnplProviders`?

**Comment by ngo275** [comment]
Also, please update README

File: Credify/Credify/Objects/PostMessageModel.swift (hunk 5 above)
Comment (lqgioan) [reply Credify/Credify/Objects/PostMessageModel.swift:36]: @ngo275 `completeBnplProviders` is correct.

**Pull request merged**